python/reduction.cpp

```diff
@@ -165,24 +165,42 @@ template <class T> Docstring docstring_bool(const std::string op) {
       .param("dim", "Dimension to reduce.", "Dim");
 }
 
+template <class T> Docstring docstring_bool(const std::string op) {
+  return Docstring()
+      .description("Element-wise " + op +
+                   " over all of the input's dimensions.")
+      .raises("If the dtype has no " + op + ", e.g., if it is a string.")
+      .returns("The " + op + " combination of the input values.")
+      .rtype<T>()
+      .template param<T>("x", "Data to reduce.");
+}
+
```

Comment:
I suggest to use the same docstring for both and write something like: `.param("dim", "Dimension to reduce (optional).", "Dim");`, and not in the `description` that all dimensions are reduced unless `dim` is given.